sdk/core/System.ClientModel/README.md

```diff
@@ -25,67 +24,144 @@ None needed for `System.ClientModel`.
 
-PipelineRequest request = message.Request;
-request.Method = "GET";
-request.Uri = new Uri("https://www.example.com/");
-request.Headers.Add("Accept", "application/json");
+The following sample illustrates how to call a convenience method and access both the strongly-typed output model and the details of the HTTP response.
 
-pipeline.Send(message);
-Console.WriteLine(message.Response.Status);
+```C# Snippet:ClientResultTReadme
+// Create a client
+string key = Environment.GetEnvironmentVariable("MAPS_API_KEY") ?? string.Empty;
+ApiKeyCredential credential = new(key);
+MapsClient client = new(new Uri("https://atlas.microsoft.com"), credential);
+
+// Call a service method, which returns ClientResult<T>
+IPAddress ipAddress = IPAddress.Parse("2001:4898:80e8:b::189");
+ClientResult<IPAddressCountryPair> result = await client.GetCountryCodeAsync(ipAddress);
+
+// ClientResult<T> has two members:
+//
+// (1) A Value property to access the strongly-typed output
+IPAddressCountryPair value = result.Value;
+Console.WriteLine($"Country is {value.CountryRegion.IsoCode}.");
+
+// (2) A GetRawResponse method for accessing the details of the HTTP response
+PipelineResponse response = result.GetRawResponse();
+
+Console.WriteLine($"Response status code: '{response.Status}'.");
+Console.WriteLine("Response headers:");
+foreach (KeyValuePair<string, string> header in response.Headers)
+{
+    Console.WriteLine($"Name: '{header.Key}', Value: '{header.Value}'.");
+}
 ```
 
+For more information on client service methods, see [Client service method samples](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/System.ClientModel/samples/ServiceMethods.md).
+
+### Handling exceptions that result from failed requests
+
+When a service call fails, service clients throw a `ClientResultException`.  The exception exposes the HTTP status code and the details of the service response if available.
+
+```C# Snippet:ClientResultExceptionReadme
+try
+{
+    IPAddress ipAddress = IPAddress.Parse("2001:4898:80e8:b::189");
+    ClientResult<IPAddressCountryPair> result = await client.GetCountryCodeAsync(ipAddress);
+}
+// Handle exception with status code 404
+catch (ClientResultException e) when (e.Status == 404)
+{
+    // Handle not found error
+    Console.Error.WriteLine($"Error: Response failed with status code: '{e.Status}'");
+}
+```
+
+### Customizing HTTP requests
+
+Service clients expose low-level _protocol methods_ that allow callers to customize the details of HTTP requests.  Protocol methods take an optional `RequestOptions` value that allows callers to add a header to the request, or to add a policy to the client pipeline that can modify the request in any way before sending it to the service.  `RequestOptions` also allows passing a `CancellationToken` to the method.
```

Comment:
This is worded as a complete set of functionality - do we want to also mention customizing `ErrorOptions`?